src/h5web/dataset-visualizer/vis/matrix/StickyContext.tsx

```diff
@@ -0,0 +1,24 @@
+import React, { ReactNode, createContext } from 'react';
+
+interface Props {
+  stickyIndices: boolean;
+  toggleStickyIndices: () => void;
+  children: ReactNode;
+}
+
+export const StickyContext = createContext({
+  stickyIndices: true,
+  toggleStickyIndices: () => {},
+});
+
+function StickyProvider(props: Props): JSX.Element {
+  const { stickyIndices, toggleStickyIndices, children } = props;
+
+  return (
+    <StickyContext.Provider value={{ stickyIndices, toggleStickyIndices }}>
+      {children}
+    </StickyContext.Provider>
+  );
+}
+
+export default StickyProvider;
```

Comment:
@axelboc-esrf Please check if this is the implementation of `Context` and its use with `State` in `IndexedGrid` are correct. Thanks !